### PR TITLE
Qualcomm AI Engine Direct - GA Static Qwen3

### PR DESCRIPTION
**Pull request by DannyYuyang-quic**
Summary:
 - support Qwen3-0.6B
 - support Qwen3-1.7B
 - refactor HF model registration for static llama

Script
``` bash
python examples/qualcomm/oss_scripts/llama/llama.py -b build-android  -s $DEVICE -m SM8750 --prompt "I would like to learn python, could you teach me with a simple example?" --temperature 0 --model_mode hybrid --prefill_ar_len 32 --max_seq_len 128 --ptq 16a4w --decoder_model qwen3
```

Stat
ee13db20-f529-413f-95c8-b6ce4bfcb4f4

### Test plan
Note: We only run Qwen3-0.6B for CI
``` bash
python backends/qualcomm/tests/test_qnn_delegate.py -k TestExampleLLMScript.test_qwen3 --model SM8650 --build_folder build-android/ --executorch_root . -s $DEVICE --artifact ./qwen3
```

cc: @haowhsu-quic, @cccclai

### CODE REVIEW

**Comment by DannyYuyang-quic** [comment]
Hi @cccclai, 
This is a PR for Qwen3-(0.6/1.7B) using static nn.Module structure.
Please notice these models are now enabled and integrated, the accuracy is still bad.
We're actively working on improvements to address this.

Thanks!

**Review by cccclai — APPROVED** [review]
Looks good

**Comment by cccclai** [comment]
There seems conflicts, can you help resolving?

**Comment by DannyYuyang-quic** [comment]
> There seems conflicts, can you help resolving?

Done!

**Comment by cccclai** [comment]
There still seems to be merge conflict...

**Comment by kimishpatel** [comment]
so when you say these models are supported via hf, what you really mean is that you can convert hf checkpoint to be loadable in static llama definition?

**Comment by cccclai** [comment]
> so when you say these models are supported via hf, what you really mean is that you can convert hf checkpoint to be loadable in static llama definition?

Both flow will be supported (static llama and hf). The hf version is https://github.com/pytorch/executorch/pull/12333 and you left some comments before. I think they address the feedback, can you take a look again
